File: src/api/schemas/agents.py
"""
Agent 相关数据模型
"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field, field_validator


class AgentCreateRequest(BaseModel):
    """Agent 创建请求"""

    name: str = Field(..., min_length=1, max_length=255, description="Agent 名称")
    model: str = Field(..., description="使用的 LLM 模型")
    system_prompt: str = Field(..., min_length=1, description="系统提示词")
    tool_names: list[str] = Field(default_factory=list, description="可用工具列表")
    max_iterations: int = Field(default=50, gt=0, le=200, description="最大迭代次数")
    timeout: int = Field(default=600, ge=0, le=3600, description="超时时间（秒）")
    description: str | None = Field(None, description="Agent 描述")
    agent_type: str = Field(default="atomic", description="Agent 类型")
    tags: list[str] = Field(default_factory=list, description="标签")
    metadata: dict[str, Any] = Field(default_factory=dict, description="元数据")

    @field_validator("name")
    @classmethod
    def validate_name(cls, v: str) -> str:
        """验证 Agent 名称"""
        if not v or not v.strip():
            raise ValueError("Agent 名称不能为空")
        return v.strip()


class AgentUpdateRequest(BaseModel):
    """Agent 更新请求"""

    name: str | None = Field(
        None, min_length=1, max_length=255, description="Agent 名称"
    )
    model: str | None = Field(None, description="使用的 LLM 模型")
    system_prompt: str | None = Field(None, min_length=1, description="系统提示词")
    tool_names: list[str] | None = Field(None, description="可用工具列表")
    max_iterations: int | None = Field(None, gt=0, le=200, description="最大迭代次数")
    timeout: int | None = Field(None, ge=0, le=3600, description="超时时间（秒）")
    description: str | None = Field(None, description="Agent 描述")
    tags: list[str] | None = Field(None, description="标签")
    metadata: dict[str, Any] | None = Field(None, description="元数据")
    status: str | None = Field(None, description="状态")


class AgentResponse(BaseModel):
    """Agent 响应"""

    id: str = Field(..., description="Agent ID")  # 改为 str 类型，支持自定义 ID 格式
    name: str = Field(..., description="Agent 名称")
    model: str = Field(..., description="使用的 LLM 模型")
    system_prompt: str = Field(..., description="系统提示词")
    tool_names: list[str] = Field(..., description="可用工具列表")
    max_iterations: int = Field(..., description="最大迭代次数")
    timeout: int = Field(..., description="超时时间（秒）")
    description: str | None = Field(None, description="Agent 描述")
    agent_type: str = Field(..., description="Agent 类型")
    status: str = Field(..., description="状态")
    tags: list[str] = Field(default_factory=list, description="标签")
    metadata: dict[str, Any] = Field(default_factory=dict, description="元数据")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime | None = Field(None, description="更新时间")


class AgentListResponse(BaseModel):
    """Agent 列表响应"""

    items: list[AgentResponse] = Field(..., description="Agent 列表")
    total: int = Field(..., ge=0, description="总数量")
    page: int = Field(..., ge=1, description="当前页码")
    page_size: int = Field(..., ge=1, description="每页数量")

File: 0.1版本前端/src/services/api/agents.ts
/** Agent 管理 API 服务 提供 Agent 配置的增删改查接口，与后端 /api/v1/agents/* 端点对齐 */

import { API_ENDPOINTS } from '@/constants/api'
import apiClient from '@/services/api/client'
import { requestWithRetry } from '@/utils/retry'
import type { RetryOptions } from '@/utils/retry'

/** Agent 响应类型（与后端 AgentResponse 对齐） */
export interface AgentResponse {
  /** Agent ID */
  id: string
  /** Agent 名称 */
  name: string
  /** Agent 描述 */
  description?: string
  /** Agent 类型 */
  agent_type: string
  /** Agent 状态 */
  status: 'active' | 'inactive' | 'error'
  /** 使用的 LLM 模型 */
  model: string
  /** 系统提示词 */
  system_prompt?: string
  /** Agent 等级（如 "L1"） */
  level?: string
  /** 绑定的工具列表 */
  tool_names?: string[]
  /** 最大迭代次数 */
  max_iterations?: number
  /** 超时时间（秒） */
  timeout?: number
  /** 标签 */
  tags?: string[]
  /** 元数据 */
  metadata?: Record<string, unknown>
  /** 创建时间 */
  created_at: string
  /** 更新时间 */
  updated_at?: string

  /** 兼容字段 - agent_type 的别名 */
  type?: string
  /** 兼容字段 - tool_names 的别名 */
  tools?: string[]
  /** 兼容字段 - metadata 的别名 */
  config?: Record<string, unknown>
}

/** Agent 列表响应类型 */
export interface AgentListResponse {
  /** Agent 列表 */
  items: AgentResponse[]
  /** 总数量 */
  total: number
  /** 当前页码 */
  page: number
  /** 每页数量 */
  page_size: number
}

/** Agent 创建请求类型（与后端 AgentCreateRequest 对齐） */
export interface AgentCreateRequest {
  /** Agent 名称 */
  name: string
  /** 使用的 LLM 模型（必需） */
  model: string
  /** 系统提示词（必需） */
  system_prompt: string
  /** Agent 描述 */
  description?: string
  /** Agent 类型 */
  agent_type?: string
  /** Agent 等级（如 "L1"） */
  level?: string
  /** 绑定的工具列表 */
  tool_names?: string[]
  /** 最大迭代次数 */
  max_iterations?: number
  /** 超时时间（秒） */
  timeout?: number
  /** 标签 */
  tags?: string[]
  /** 元数据 */
  metadata?: Record<string, unknown>
}

/** Agent 更新请求类型（与后端 AgentUpdateRequest 对齐） */
export interface AgentUpdateRequest {
  /** Agent 名称 */
  name?: string
  /** Agent 描述 */
  description?: string
  /** Agent 类型 */
  agent_type?: string
  /** Agent 状态 */
  status?: 'active' | 'inactive'
  /** 使用的 LLM 模型 */
  model?: string
  /** 系统提示词 */
  system_prompt?: string
  /** Agent 等级（如 "L1"） */
  level?: string
  /** 绑定的工具列表 */
  tool_names?: string[]
  /** 最大迭代次数 */
  max_iterations?: number
  /** 超时时间（秒） */
  timeout?: number
  /** 标签 */
  tags?: string[]
  /** 元数据 */
  metadata?: Record<string, unknown>
}

/** 获取 Agent 列表查询参数 */
export interface GetAgentsParams {
  /** 页码 */
  page?: number
  /** 每页数量 */
  pageSize?: number
  /** 状态过滤 */
  status?: string
  /** 类型过滤 */
  type?: string
  /** 搜索关键词 */
  search?: string
}

export async function getAgents(
  params: GetAgentsParams = {},
  options: RetryOptions = {},
): Promise<AgentListResponse> {
  return requestWithRetry(async () => {
    const response = await apiClient.get<AgentListResponse>(API_ENDPOINTS.AGENTS.LIST, {
      params: {
        page: params.page || 1,
        page_size: params.pageSize || 20,
        status: params.status,
        agent_type: params.type,
        search: params.search,
      },
    })
    return response.data
  }, options)
}

export async function getAgent(
  agentId: string,
  options: RetryOptions = {},
): Promise<AgentResponse> {
  if (!agentId || agentId.trim().length === 0) {
    throw new Error('Agent ID 不能为空')
  }

  return requestWithRetry(async () => {
    const response = await apiClient.get<AgentResponse>(API_ENDPOINTS.AGENTS.GET(agentId))
    return response.data
  }, options)
}

export async function createAgent(
  data: AgentCreateRequest,
  options: RetryOptions = {},
): Promise<AgentResponse> {
  if (!data.name || data.name.trim().length === 0) {
    throw new Error('Agent 名称不能为空')
  }

  return requestWithRetry(async () => {
    const response = await apiClient.post<AgentResponse>(API_ENDPOINTS.AGENTS.CREATE, data)
    return response.data
  }, options)
}

export async function updateAgent(
  agentId: string,
  data: AgentUpdateRequest,
  options: RetryOptions = {},
): Promise<AgentResponse> {
  if (!agentId || agentId.trim().length === 0) {
    throw new Error('Agent ID 不能为空')
  }

  return requestWithRetry(async () => {
    const response = await apiClient.put<AgentResponse>(API_ENDPOINTS.AGENTS.UPDATE(agentId), data)
    return response.data
  }, options)
}

export async function deleteAgent(agentId: string, options: RetryOptions = {}): Promise<void> {
  if (!agentId || agentId.trim().length === 0) {
    throw new Error('Agent ID 不能为空')
  }

  return requestWithRetry(async () => {
    await apiClient.delete(API_ENDPOINTS.AGENTS.DELETE(agentId))
  }, options)
}

export async function getDefaultAgent(options: RetryOptions = {}): Promise<AgentResponse> {
  return requestWithRetry(async () => {
    const response = await apiClient.get<AgentResponse>(API_ENDPOINTS.AGENTS.DEFAULT)
    return response.data
  }, options)
}
